lucene/core/src/java/org/apache/lucene/codecs/lucene912/Lucene912PostingsReader.java

```diff
@@ -1529,36 +1498,27 @@ public int getDocIdUpTo(int level) {
             level--;
           }
 
-          if (level1LastDocID != NO_MORE_DOCS) {
-            if (level == 0) {
-              return level1LastDocID;
-            }
-            level--;
+          if (level1LastDocID != NO_MORE_DOCS && level == 0) {
```

Comment:
No need to check `level1LastDocID` against `NO_MORE_DOCS` since we return `NO_MORE_DOCS` otherwise anyway?

```suggestion
          if (level == 0) {
```